### ISSUE TITLE
NVLink causing failure of "docker run --gpus all nvidia/cuda:10.0-base nvidia-smi"

### ISSUE THREAD

**Issue by markmp** [NVIDIA/nvidia-docker#1180]
An Ubuntu upgrade hurt my life. Also of a sudden couldn't boot after update. So I had to purge the driver and reinstall. Finally got video back and driver seemed to be working fine outside of containers. But then I couldn't use Nvidia NGC containers anymore. After upgrading my kernel to 5.3.18 and Nvidia driver 440, I kept getting this error: 

docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused "process_linux.go:432: running prestart hook 1 caused "error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: cuda error: no cuda-capable device is detected\n""": unknown.
ERRO[0003] error waiting for container: context canceled

I uninstalled and reinstalled docker, Nvidia driver (430, 435, 440), multiple kernels (5.0.0, 5.3.18, 4.x.x) many many times as some of these threads suggested. Nothing worked and I've been stuck on this all day.

Then I found one post pointing to Nvlink as the problem. As a last hope before giving up, I opened my case ripped my Nvlink out of my system. Solved! Unreal. Somehow the physical Nvlink which used to work, no longer functions with 435, 440 and a 5+ kernel.

**Comment by bluebox42** [comment]
See also https://github.com/NVIDIA/nvidia-docker/issues/1070

**Comment by RenaudWasTaken** [comment]
Hello!

Can you run `nvidia-bug-report.sh` with the nvlink connector on and post the results here?

Thanks!

**Comment by bluebox42** [comment]
At the moment i do not have physical acces to the system.

@markmp can you post the output of nvidia-bug-report.sh?

**Comment by markmp** [comment]
Ok - I ran the script and attached the full log. But of course, now I can't seem to replicate the error. I didn't make any software or other changes since my last post. Only thing I did was open the case back up and put the NVlink back on between GPUs 0 and GPU 2 (there are three total), and now the error is gone. Only thing I can think of is that having NVlink on during a fresh install might have caused the issue. I wonder if there is a weird corner case of doing the driver and container kit install after the physical NVlink was attached. Will post again if I can replicate, but I don't want to mess with it now to test that hypothesis. 

[nvidia-bug-report.log.gz](https://github.com/NVIDIA/nvidia-docker/files/4132163/nvidia-bug-report.log.gz)
